drivers/hmis/app/models/hmis/project_unit_type_mapping.rb

```diff
@@ -0,0 +1,58 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# exported data from ACHMIS that describes the unit types and capacity for each project (called program)
+class Hmis::ProjectUnitTypeMapping < Hmis::HmisBase
+  self.table_name = :hmis_project_unit_type_mappings
+
+  belongs_to :project, class_name: 'Hmis::Hud::Project'
+  belongs_to :unit_type, class_name: 'Hmis::UnitType'
+
+  scope :active, -> { where(active: true) }
+
+  def self.freshen_project_units(user:, today: Date.current)
+    scope = preload(:project, :unit_type)
+    # { [project_id, unit_type_id] => unit_count }
+    unit_counts_by_project_and_unit_type_id = Hmis::Unit
+      .where(project_id: scope.map { |r| r.project.id })
+      .group(:project_id, :unit_type_id)
+      .count
```

Comment:
I'm not following this. why this can't just be:

```suggestion
    # { [project_id, unit_type_id] => unit_count }
    unit_counts_by_project_and_unit_type_id = Hmis::Unit.group(:project_id, :unit_type_id).count
```

Reply:
My thought was to restrict this count/group to just the projects included in the mappings (rather than all units for all projects). Maybe that's not worth the extra bother though, I could buy that. I'll take out the condition